src/hooks/useComponentShow.ts

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect, useRef } from 'react';
+
+export const useComponentVisible = (initialIsVisible: boolean) => {
+  const [isComponentVisible, setIsComponentVisible] =
+    useState(initialIsVisible);
+  const ref = useRef<HTMLButtonElement>(null);
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (ref.current && !ref.current.contains(event.target as Node)) {
+      setIsComponentVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClickOutside, true);
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true);
+    };
+  }, []);
+
+  return { ref, isComponentVisible, setIsComponentVisible };
+};
```

Comment:
Create another hook that listens to keyboard-interactions. Use it to set the component invisible if Escape is pressed. Then create a test case for it